src/core/public/map_to_object.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function mapToObject(map: Map<string, unknown>) {
```

Comment:
I'd prefer this file to exist in `src/core/utils` rather than litter the top-level `public` directory of core.

Reply:
I can move, np. afair we were going to remove this folder. aren't we?

Reply:
as for me I'm fine to keep this folder for such env-agnostic utils

Reply:
We discussed real quick and decided we're fine with the `src/core/utils` directory existing for only _very generic vanilla JS_ utilities similar to what you'd find in lodash or a standard library.